src/change_stream.ts

```diff
@@ -668,18 +659,30 @@ export class ChangeStream<
   next(callback: Callback<TChange>): void;
   next(callback?: Callback<TChange>): Promise<TChange> | void {
     this._setIsIterator();
-    return maybePromise(callback, cb => {
-      this._getCursor((err, cursor) => {
-        if (err || !cursor) return cb(err); // failed to resume, raise an error
-        cursor.next((error, change) => {
-          if (error) {
-            this[kResumeQueue].push(() => this.next(cb));
-            this._processError(error, cb);
-            return;
+    // TOOD(NODE-4319): Add eslint rule preventing accidental variable shadowing
+    // Shadowing is intentional here.  We want to override the `callback` variable
+    // from the outer scope so that the inner scope doesn't accidentally call the wrong callback.
+    return maybePromise(callback, callback => {
+      (async () => {
+        try {
+          const change = await this.cursor.next();
+          const processedChange = this._processChange(change ?? null);
+          return processedChange;
+        } catch (error) {
+          try {
+            await this._processErrorIteratorMode(error);
+            const change = await this.cursor.next();
+            const processedChange = this._processChange(change ?? null);
+            return processedChange;
+          } catch (error) {
+            await this.close().catch(err => err);
+            throw error;
           }
```

Comment:
```suggestion
          } finally {
            await this.close();
          }
```
I'm not seeing where .close will throw or reject, so I'm not sure we should attach a silencer, thoughts? also catching then throwing seems like a use case for finally